ledger/eval.go

```diff
@@ -1171,22 +1163,35 @@ func eval(ctx context.Context, l ledgerForEvaluator, blk bookkeeping.Block, vali
 
 	}
 
-	for _, txgroup := range paysetgroups {
+	base := eval.state.lookupParent.(*roundCowBase)
+
+	for {
 		select {
+		case txgroup, ok := <-paysetgroupsCh:
+			if !ok {
+				break
+			} else if txgroup.err != nil {
+				return ledgercore.StateDelta{}, err
+			}
+
+			for _, br := range txgroup.balances {
+				base.accounts[br.Addr] = br.AccountData
+			}
+			err = eval.TransactionGroup(txgroup.group)
+			if err != nil {
+				return ledgercore.StateDelta{}, err
+			}
+			continue
 		case <-ctx.Done():
 			return ledgercore.StateDelta{}, ctx.Err()
 		case err, open := <-txvalidator.done:
 			// if we're not validating, then `txvalidator.done` would be nil, in which case this case statement would never be executed.
 			if open && err != nil {
 				return ledgercore.StateDelta{}, err
 			}
-		default:
-		}
-
-		err = eval.TransactionGroup(txgroup)
-		if err != nil {
-			return ledgercore.StateDelta{}, err
+			continue
 		}
+		break
```

Comment:
I needed this because of the lines 1171-1172 : 
in case we're running out of transaction groups in the channel, we need to exit the for loop. Using "break" won't exit the for loop since we're in a select statement.

Reply:
I agree with Brian - add `break outer` - see suggested change.